### PR TITLE
Refactoring related to Selenium 4 API deprecations

### PR DESCRIPTION
**Pull request by quarckster**
- Added type annotations
- `webdriver_options` are converted to `selenium.webdriver.ChromeOptions` or `selenium.webdriver.FirefoxOptions`
- `BrowserFactory` passes only allowed arguments to the webdriver
- Tests use `podman` in a subprocess to run a container with selenium
- Added `test_session`
- Selenium 4.0 is minimal version
- `setuptools` is replaced with `hatch`
- `setup.cfg` is replaced with `pyproject.toml`
- `.pre-commit-config.yaml` updated
- `flake8` config is moved to `pre-commit`
- `.isort.cfg` is deleted
- Updated Github Actions config

### CODE REVIEW

File: src/webdriver_kaifuku/__init__.py [hunk 1]
```diff
@@ -105,32 +68,14 @@ def create(self):
...
         if browser:
             browser.quit()
 
 
-@attr.s
+@attr.s(auto_attribs=True)
 class WharfFactory(BrowserFactory):
-    wharf = attr.ib()
-
-    DEFAULT_WHARF_CHROME_OPT_ARGS = ["--no-sandbox"]
```
Comment (RonnyPfannschmidt) [inline src/webdriver_kaifuku/__init__.py:117]: Does this affect cfme?

**Review by RonnyPfannschmidt — COMMENTED** [review]
this is a massive squashed change that takes time to review, so next week when im back from pto

File: src/webdriver_kaifuku/__init__.py (hunk 1 above)
Comment (quarckster) [reply src/webdriver_kaifuku/__init__.py:117]: I'm not aware if cfme is still used. Anyway, I would drop wharf support at all.

**Comment by quarckster** [comment]
> this is a massive squashed change that takes time to review, so next week when im back from pto

I'll split it into meaningful commits.

File: src/webdriver_kaifuku/__init__.py (hunk 1 above)
Comment (RonnyPfannschmidt) [reply src/webdriver_kaifuku/__init__.py:117]: drop is ok

File: testing/conftest.py [hunk 2]
```diff
@@ -1,120 +1,77 @@
...
+from urllib.request import urlopen
 
-import docker
 import pytest
 from wait_for import wait_for
 
-WHARF_URL = "http://localhost:4899"
-WHARF_IMAGE = "cfmeqe/webdriver-wharf"
-BROWSER_IMAGE = "cfmeqe/cfme_sel_stable"
+BROWSER_IMAGE = "quay.io/redhatqe/selenium-standalone:latest"
```
Comment (RonnyPfannschmidt) [inline testing/conftest.py:8]: followup note - lets make this something the integration has to supplement

**Review by RonnyPfannschmidt — APPROVED** [review]

**Pull request merged**